### PR TITLE
Update `activate_this.py`; avoid polluting `__main__`

### PR DESCRIPTION
**Pull request by t-kalinowski**
## Summary

Currently, running `activate_this.py` leaves the following symbols in `__main__`:
```
abs_file annotations base bin_dir lib os path prev_length site sys
```
This PR updates the `activate_this.py` script so it does not leave any detritus in `__main__`.

## Test Plan

This could be tested with something like this:

```bash
activate_this=$(python -c '__import__("sys").exec_prefix + "/bin/activate_this.py"')
pollution=$(python -c "start = set(dir()); __import__('runpy').run_path($activate_this); print(start.difference(['start']+dir()))'

# if len(pollution), error
```

### CODE REVIEW

File: crates/uv-virtualenv/src/activator/activate_this.py [hunk 1]
```diff
@@ -28,32 +28,41 @@
...
-    path = os.path.realpath(os.path.join(bin_dir, lib))
-    site.addsitedir(path)
-sys.path[:] = sys.path[prev_length:] + sys.path[0:prev_length]
-
-sys.real_prefix = sys.prefix
-sys.prefix = base
+
+def _activate_this():
+
+    from __future__ import annotations
```
Comment (bluss) [inline crates/uv-virtualenv/src/activator/activate_this.py:34]: will this run?

Comment (j178) [inline crates/uv-virtualenv/src/activator/activate_this.py:34]: Is this valid?

https://docs.python.org/3/reference/simple_stmts.html#future 

> A future statement must appear near the top of the module.

Comment (t-kalinowski) [reply crates/uv-virtualenv/src/activator/activate_this.py]: Good catch! I removed it (it appears to be unused in the file).

**Comment by konstin** [comment]
Hi! I've tried the reproducer command, but for me

```console
python -c "start = set(dir()); __import__('runpy').run_path('.venv/bin/activate_this.py'); print(start.difference(['start']+dir()))"
```

returns an empty set. I'm probably using a different setup though since `activate_this=$(python -c '__import__("sys").exec_prefix + "/bin/activate_this.py"')` resolves to an empty string in my bash on ubuntu 22.04 (`sys.exec_prefix` is `/usr` fwiw), could you tell me some more about your setup?

**Comment by t-kalinowski** [comment]
Thanks. I see now that runpy does some additional isolation and doesn't execute directly in `__main__`.

In reticulate, we run the script using the Python C API: `PyRun_StringFlags()` (https://github.com/rstudio/reticulate/blob/d2c83b04ebc6fbecab057c369c2562adb332db67/src/python.cpp#L2749). 

Perhaps running the activate script in a new dict is more correct, 🤔. Still, that's old, and very exercised code, and this is the first time I've seen this issue.

**Comment by t-kalinowski** [comment]
But I see that `virtualenv` also takes a similar approach as here, https://github.com/pypa/virtualenv/blob/main/src/virtualenv/activation/python/activate_this.py. I'm starting to think the fix needs to be in reticulate.

**Pull request closed without merging**